addons/website/views/website_views.xml

```diff
@@ -298,6 +303,95 @@
+                            <div class="input-group-append">
+                                <button type="submit" class="btn btn-info" aria-label="Search" title="Search">
+                                    <i class="fa fa-search"/>
+                                </button>
+                            </div>
+                        </div>
+                    </nav>
+
+                    <t t-set="view" t-value="env['ir.ui.view'].browse(context['active_id']).with_context(active_test=False)"/>
+                    <t t-set="top_level_view" t-value="view._get_top_level_view()"/>
+                    <div class="o_tree_container ml-2">
+                        <!-- TODO those 2 <a> should load tree hierarchy, not form view -->
+                        <div t-if="top_level_view != view" class="alert alert-info m-1 p-1">
+                            This view is not a top level view
+                            <a type="action" data-model="ir.ui.view" t-att-data-res-id="top_level_view.id" href="#">
+                                <i class="fa fa-arrow-right mr-1"/>Show whole hierarchy
+                            </a>
+                        </div>
+                        <t t-set="sibling_views" t-value="view.search([('key', '=', view.key)]) - view"/>
+                        <div t-if="sibling_views" class="alert alert-info m-1 p-1">
+                            Multiple tree exists for this view
+                            <a t-foreach="sibling_views" t-as="sibling_view" type="action" data-model="ir.ui.view" t-att-data-res-id="sibling_view.id" href="#">
+                                <i class="fa fa-arrow-right mr-1"/>
+                                <t t-esc="sibling_view.with_context(display_website=True).display_name"/>
+                            </a>
+                        </div>
+                        <t t-call="website.report_viewhierarchy_children"/>
+                    </div>
+                </qweb>
+            </field>
+        </record>
+        <template id="report_viewhierarchy_children">
+            <t t-set="classes_for_search" t-value="'d-flex align-items-center'"/>
+            <p t-attf-class="o_tree_entry mb-0 #{view.active and 'text-info' or 'text-muted is_inactive'} #{view.inherit_children_ids and 'o_has_child' or ''} #{classes_for_search}">
+                <i t-if="view.inherit_children_ids" class="js_fold o_fold_icon fa fa-plus-circle mr-1"/>
+                <i t-if="view.arch_updated" class="fa fa-pencil mr-1"/>
+                <span class="js_fold" t-esc="view.with_context(display_website=True).display_name"/>
+                <a type="action" data-model="ir.ui.view" t-att-data-res-id="view.id">
+                    <i class="fa fa-eye ml-2" title="Go to View"/>
+                </a>
+                <!-- TODO button to show diff -->
+                <a type="action" name="base.reset_view_arch_wizard_action"
```

Comment:
Not too sure yet how to pass active_model and active_ids to the python action, despite the context attribute, it is empty on the python action